product_search/product_search.py

```diff
@@ -0,0 +1,41 @@
+"""Program that returns inventory details based for an inventory item"""
+
+
+def get_product_name_from_user():
+    """Get a name of a product from the user"""
+    return input("What is the product name? ").title()
+
+
+def item_inventory_details(item_to_search, data):
+    """Verify that the product exists; if no record is found, prompt again."""
+
+    while True:
+        for item in data["products"]:
+            if item["name"] == item_to_search:
+                return item
+        print("Sorry, that product was not found in our inventory.")
+        item_to_search = get_product_name_from_user()
+
+
+def _style_dictionary_items_for_output(inv_details):
+    """Update keys, values in inventory dictionary for user readability"""
+    user_inventory_items = {}
+
+    for category, details in inv_details.items():
+        if category == "quantity":
+            category = "Quantity on hand"
+        if category == "price":
+            details = "${0:.2f}".format(inv_details["price"])
+        user_inventory_items[category] = details
+    return user_inventory_items
```

Comment:
@paulghaddad Is this a place to use reduce?

Reply:
I think this would be a difficult reduce.